snapcraft/cli/remote.py

```diff
@@ -0,0 +1,245 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2019 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import click
+import os
+import uuid
+
+from snapcraft.internal.remote_build import Worktree, LaunchpadClient, Repo, errors
+from typing import List, Tuple
+from xdg import BaseDirectory
+from . import echo
+from ._options import get_project
+from snapcraft import InfoFile
+
+_SUPPORTED_ARCHS = ["amd64", "arm64", "armhf", "i386", "ppc64el", "s390x"]
```

Comment:
I have this same structure for the details to info migration....
Maybe this should go into storeapi.constants

Reply:
and made public

Reply:
I propose we land this first, and then refactor to use a common infrastructure for both cases.